primitives/core/src/crypto.rs

```diff
@@ -529,6 +530,76 @@ pub trait Public: AsRef<[u8]> + AsMut<[u8]> + Default + Derive + CryptoType + Pa
 	fn as_slice(&self) -> &[u8] { self.as_ref() }
 }
 
+/// Trait suitable for typical cryptographic PKI signature type.
+pub trait Signature: AsRef<[u8]> + CryptoType + PartialEq + Eq + Clone + Send + Sync {
+	/// A new instance from the given slice.
+	fn from_slice(data: &[u8]) -> Self;
+
+	/// Return a `Vec<u8>` filled with raw data.
+	fn to_raw_vec(&self) -> Vec<u8> { self.as_slice().to_vec() }
+
+	/// Return a slice filled with raw data.
+	fn as_slice(&self) -> &[u8] { self.as_ref() }
+}
+
+/// An enum which holds references to public keys which are
+/// supported for signing messages
+#[cfg(feature = "full_crypto")]
+pub enum SupportedPublicKey {
```

Comment:
Is it possible to use `AppKey` directly and drop this and the related enum in favor of it?

Reply:
@gnunicorn and I have discussed this and here's a summary of our thoughts about this:

`AppPair`/`AppPublic` ... etc are defined in `sp-app-crypto` while the `BareCryptoStore` trait is defined in `sp-core`. The former crate depends on the latter so it's not a good idea to `use sp_app_crypto` inside `sp_core`. Therefore, the trait cannot use `AppPublic` as a parameter type in replacement of `SupportedPublicKey`.